### ISSUE TITLE
Multiple files upload

### ISSUE THREAD

**Issue by heinbez** [Upload/Up1#46]
Hi,

Thanks for building this project. A feature that I would really appreciate is being able up upload multiple files at once.

Thanks,
H.

**Comment by andre-d** [comment]
#4

**Comment by k3d3** [comment]
I don't think this is the same as #4 - I think he just wants to upload multiple files at once, not necessarily put those files into an album. 

Still, adding multiple file upload and then putting a checkbox somewhere stating "create an album" would solve both of them.

**Comment by andre-d** [comment]
Not the same, if it were the same I would have closed it, this is just a
flaw in the way github relates issues (you need to make a comment).   Just
related, the "here are the files you have uploaded" view is the same as the
album view minus actually uploading the index and both depend on uploading
multiple at once.

**Comment by k3d3** [comment]
Fair enough, I've modified your comment. Feel free to change it back if you prefer

**Comment by perpi** [comment]
@k3d3, you mean sth like imgur.com? doesn't it need user profiles?!

**Comment by andre-d** [comment]
No, albums will not require accounts.  Basically the server will be
modified to allow you to overwrite if you have the deletion key which will
allow you to overwrite an album index as you add to it linking to other
files in one index.

There will also be albums via + in the URL.

**Comment by k3d3** [comment]
Nope, you wouldn't need user profiles. For an album, basically you would just upload multiple files normally, and then create a new "upload" that just contains an table of contents for the just-uploaded files.

**Comment by heinbez** [comment]
Is up1 a no database solution? Could you potentially just use folders to create a collection of items?

**Comment by andre-d** [comment]
There is no database except the filesystem

**Comment by k3d3** [comment]
@heinbez yep, there's no database involved - just the file storage on the filesystem. That way it can easily be scaled out via FS-scaling solutions like glusterfs.

We'd prefer having a single flat directory, as it makes it easier to just use a plain static web server like nginx to serve the files. Having a directory would require weird things like directory listings in collections directories and not anywhere else. We want the server to be as dumb as possible.

For multiple file uploads, we plan on giving the option to either 1) giving multiple individual links, or 2) creating an album file that refers to the aforementioned multiple individual links. This also has a nice side-effect of being able to refer to existing images and files in new albums.

**Comment by heinbez** [comment]
Cool! Looking forward to the update :)

**Comment by elimisteve** [comment]
> For multiple file uploads, we plan on giving the option to either 1) giving multiple individual links, or 2) creating an album file that refers to the aforementioned multiple individual links. This also has a nice side-effect of being able to refer to existing images and files in new albums.

@andre-d @k3d3 I came here hoping to find something like this!  Are there plans to add this feature? And either way, is there a design laid out somewhere.  I really want this feature and may be able to add it.

**Comment by wolfgangasdf** [comment]
I modified up1 (too much for a fork, mainly for personal use) including multi-file upload (browser ZIP, not focussed on pictures), probably this is useful for someone: https://github.com/wolfgangasdf/up1234me
